include/libpldm/cpp/firmware_update.hpp

```
@@ -0,0 +1,200 @@
+
+	struct FirmwareDeviceIDRecord
+		: GrowableStruct<struct FirmwareDeviceIDRecord>,
+		  private NonCopyableNonMoveable {
+	    private:
+		friend pldm::fw_update::PackageParser;
+
+		FirmwareDeviceIDRecord(
+			const std::bitset<32> &deviceUpdateOptionFlags,
+			const std::vector<size_t> &applicableComponents,
+			const std::string &componentImageSetVersionString,
+			const std::map<uint16_t, std::unique_ptr<DescriptorData> >
```

Clang (14.0.6)

[error] parameter 4 is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions

(readability-avoid-const-params-in-decls,-warnings-as-errors)